src/solutions/pde_solutions.jl

```diff
@@ -0,0 +1,190 @@
+    u::uType
+    original_sol::Sol
+    errors::DType
+    t::tType
+    ivgrid::gridType
+    ivs::ivType
+    dvs::dvType
+    disc_data::Disc
+    prob::P
+    alg::A
+    interp::IType
+    dense::Bool
+    tslocation::Int
+    retcode::Symbol
+end
+
+function PDESolution(sol::ODESolution{T}, metadata::MOLMetadata) where {T}
+    odesys = sol.prob.f.sys
+
+    pdesys = metadata.pdesys
+    discretespace = metadata.discretespace
+
+    ivs = [discretespace.time, discretespace.x̄...]
+    ivgrid = (isequal(discretespace.time, x) ? sol.t : discretespace.grid[x] for x in ivs)
+    # Reshape the solution to flat arrays
+    umap = Dict(map(discretespace.ū) do u
+                    let discu = discretespace.discvars[u]
+                        solu = map(CartesianIndices(discu)) do I
+                            i = sym_to_index(discu[I], odesys)
+                            # Handle Observed
+                            if i !== nothing
+                                sol.u[i]
+                            else
+                                observed(sol, discu[I], :)
+                            end
+                        end
+                        out = zeros(T, length(sol.t), size(discu)...)
+                        for I in CartesianIndices(discu)
+                            out[:, I] .= solu[I]
+                        end
+                        u => out
+                    end
+                end)
+    # Build Interpolations
+    # TODO: Check if the grid is uniform and use the supported higher order interpolations
+    interp = Dict(map(keys(umap)) do k
+                      args = arguments(k) #! Do without Symbolics
+                      nodes = Tuple((map(args) do arg
+                                         i = findfirst(arg, ivs)
+                                         @assert i!==nothing "Independent variable $arg
+                                                              not found in ivs"
+                                         ivgrid[i]
+                                     end)...)
+                      k => interpolate(nodes, umap[k], Gridded(Linear()))
+                  end)
+
+    return PDESolution{T, length(discretespace.ū), typeof(umap), typeof(metadata),
+                       typeof(sol), typeof(sol.errors), typeof(sol.t), typeof(ivgrid),
+                       typeof(ivs), typeof(pdesys.dvs), typeof(sol.prob), typeof(sol.alg),
+                       typeof(interp)}(umap, sol, sol.errors, sol.t, ivgrid, ivs,
+                                       pdesys.dvs, metadata, sol.prob, sol.alg,
+                                       interp, sol.dense, sol.tslocation,
+                                       sol.retcode)
+end
+
+wrap_sol(sol::ODESolution, disc_data::MOLMetadata) = PDESolution(sol, disc_data)
+
+function (sol::PDESolution{T, N, S, D})(args...;
+                                        dv = nothing) where {T, N, S, D <: MOLMetadata}
```

Comment:
This should just lower to a general metadata dispatch which would then be implemented in each PDE solver package. That would then move MOLMetadata to MethodOfLines.jl, and allow for NeuralPDEMetadata in NeuralPDE.jl